src/librustc_mir/borrow_check/nll/type_check/input_output.rs

```diff
@@ -50,18 +50,18 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
         // Equate expected input tys with those in the MIR.
         let argument_locals = (1..).map(Local::new);
         for (&unnormalized_input_ty, local) in unnormalized_input_tys.iter().zip(argument_locals) {
-            let input_ty = self.normalize(&unnormalized_input_ty, Locations::All);
+            let input_ty = self.normalize(&unnormalized_input_ty, Locations::Entry);
             let mir_input_ty = mir.local_decls[local].ty;
-            self.equate_normalized_input_or_output(input_ty, mir_input_ty);
+            self.equate_normalized_input_or_output(Locations::Entry, input_ty, mir_input_ty);
         }
 
         assert!(
-            mir.yield_ty.is_some() && universal_regions.yield_ty.is_some() ||
-            mir.yield_ty.is_none() && universal_regions.yield_ty.is_none()
+            mir.yield_ty.is_some() && universal_regions.yield_ty.is_some()
+                || mir.yield_ty.is_none() && universal_regions.yield_ty.is_none()
         );
         if let Some(mir_yield_ty) = mir.yield_ty {
             let ur_yield_ty = universal_regions.yield_ty.unwrap();
-            self.equate_normalized_input_or_output(ur_yield_ty, mir_yield_ty);
+            self.equate_normalized_input_or_output(Locations::Yield, ur_yield_ty, mir_yield_ty);
```

Comment:
I think I need a comment here saying *this* is the place where effectively you are equating at all points. (or am I even right about that?)

Reply:
@nikomatsakis still would like to see a comment in the source code about the effect of this change. (Either my hypothesis is right, which means its a subtle effect. Or my hypothesis is wrong, which means I need to figure out where the equate-at-all-points **is** happening for Yield...)